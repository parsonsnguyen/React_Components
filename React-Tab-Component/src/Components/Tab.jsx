import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Context from '../tab-panel-context';

export default class Tab extends Component {
    render() {
        const { tabIndex, tabHeader } = this.props;
        return (
            <Context.Consumer>{
                ({ onSelectedTabIndexChange, selectedTabIndex }) =>
                    <li className={`la-tabs-nav-list-item ${selectedTabIndex === tabIndex ? "isSelected" : ""}`}>
                        <a className={`la-tabs-nav-link`} onClick={() => onSelectedTabIndexChange(tabIndex)}>{tabHeader}</a>
                    </li>
            }</Context.Consumer>);
    }
};

Tab.defaultProps = {
    tabHeader: ""
}

Tab.propTypes = {
    tabHeader: PropTypes.string,
    tabIndex: PropTypes.number
}