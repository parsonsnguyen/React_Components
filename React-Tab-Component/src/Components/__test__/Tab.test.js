import React from 'react'
import Tab from '../Tab'
import Context from '../../tab-panel-context'
import rerender from 'react-test-renderer'

describe('Tab: ', () => {
    test('renders correctly', () => {
        const component = rerender
            .create(
                <Context.Provider value={({ onSelectedTabIndexChange: jest.fn() })}>
                    <Tab tabIndex={0} tabHeader="tab-header">Content</Tab>
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    test('has selectedTabIndex equals to tabIndex: className should contain isSelected value', () => {
        const dumnmyIndex = 999;
        const component = rerender
            .create(
                <Context.Provider value={({ onSelectedTabIndexChange: jest.fn(), selectedTabIndex: dumnmyIndex })}>
                    <Tab tabIndex={dumnmyIndex} tabHeader="tab-header">Content</Tab>
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test.skip('onSelectedTabIndexChange()', () => {
        const onSelectedTabIndexChange = jest.fn();
        const component = rerender
            .create(
                <Context.Provider value={({ onSelectedTabIndexChange: onSelectedTabIndexChange })}>
                    <Tab tabIndex={0} tabHeader="tab-header">Content</Tab>
                </Context.Provider>
            );
        console.log(component.toJSON().children[0].props);
        expect(component.toJSON().children[0].props.onClick).toBe(onSelectedTabIndexChange);
    });
});
