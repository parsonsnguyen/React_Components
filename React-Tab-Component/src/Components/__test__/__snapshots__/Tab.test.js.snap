// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab:  has selectedTabIndex equals to tabIndex: className should contain isSelected value 1`] = `
<li
  className="la-tabs-nav-list-item isSelected"
>
  <a
    className="la-tabs-nav-link"
    onClick={[Function]}
  >
    tab-header
  </a>
</li>
`;

exports[`Tab:  renders correctly 1`] = `
<li
  className="la-tabs-nav-list-item "
>
  <a
    className="la-tabs-nav-link"
    onClick={[Function]}
  >
    tab-header
  </a>
</li>
`;
